mongomock/filtering.py

```diff
@@ -154,6 +154,13 @@ def _regex(doc_val, regex):
     return any(regex.search(item) for item in _force_list(doc_val))
 
 
+def _size_op(doc_val, search_val):
+    if isinstance(doc_val, (list, tuple, dict)):
+        return search_val == len(doc_val)
+    else:
+        return search_val == 1 if doc_val else 0
```

Comment:
Somewhere on this conversation we've forget what the change is about. This change supposedly should check if an array of items is as long as we expect with $size operator.

If item isinstance of an iterable type(list, tuple, dict) it should check it's length, else, it's size is based in if it exists or not (1 or 0, that doesn't mean True or False, it is it's length).
So, I'd recommend this change:
```
+    if isinstance(doc_val, (list, tuple, dict)):
+        return search_val == len(doc_val)
+    return search_val == 1 if doc_val else 0
```